pkg/reaper/podreaper/podreaper.go

```diff
@@ -29,7 +32,7 @@ import (
 var log = logrus.New()
 
 // Run is the main runner function for pod-reaper, and will initialize and start the pod-reaper
-func Run(args *Args) error {
+func Run(args *Args, fakePods ...[]FakePod) error {
```

Comment:
We should not make this change, adding faking at this level is not a correct refactor in my opinion.
Run should simply take in `kubernetes.Interface` and a fake client could be passed in at a higher level

Reply:
The problem here is that a Reaper Context is built in the Run function. Could we maybe pass the context in its entirety instead?

Reply:
@eytan-avisror

Reply:
I agree that we should keep mocks only in testing. How would we bypass `validateArguments` in this scenario.